### PR TITLE
Add C-Reduce and dependencies.

### PR DESCRIPTION
**Pull request by olupton**
This PR adds a recipe for C-Reduce: https://embed.cs.utah.edu/creduce/
> C-Reduce is a tool that takes a large C, C++, or OpenCL file that has a property of interest (such as triggering a compiler bug) and automatically produces a much smaller C/C++ file that has the same property. It is intended for use by people who discover and report bugs in compilers and other tools that process source code.

It also adds recipes for some Perl dependencies:
 - `perl-exporter-lite`
 - `perl-getopt-tabular`

### CODE REVIEW

File: var/spack/repos/builtin/packages/creduce/package.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+##############################################################################
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/creduce/package.py]:
```suggestion
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Only a minor request. I'll ping @glennpj to have a look at perl packages.

File: var/spack/repos/builtin/packages/creduce/package.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    depends_on('flex')
+    depends_on('libxml2')
+    depends_on('llvm@8.0.0', when='@:2.10.999')
+    depends_on('perl')
+    depends_on('perl-exporter-lite')
+    depends_on('perl-file-which')
+    depends_on('perl-getopt-tabular')
+    depends_on('perl-regexp-common')
+    depends_on('perl-term-readkey')
```
Comment (glennpj) [inline var/spack/repos/builtin/packages/creduce/package.py]: This package already exists in spack as `perl-termreadkey`.
```suggestion
    depends_on('perl-termreadkey')
```

File: var/spack/repos/builtin/packages/perl-term-readkey/package.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+class PerlTermReadkey(PerlPackage):
+    """Term::ReadKey is a compiled perl module dedicated to providing simple
+    control over terminal driver modes (cbreak, raw, cooked, etc.,) support
+    for non-blocking reads, if the architecture allows, and some generalized
+    handy functions for working with terminals."""
+    homepage = "https://metacpan.org/pod/Term::ReadKey"
+    url      = "https://cpan.metacpan.org/authors/id/J/JS/JSTOWE/TermReadKey-2.38.tar.gz"
+
+    version('2.38', sha256='5a645878dc570ac33661581fbb090ff24ebce17d43ea53fd22e105a856a47290')
```
Comment (glennpj) [inline var/spack/repos/builtin/packages/perl-term-readkey/package.py]: This package should be removed from the PR.

**Review by glennpj — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/perl-term-readkey/package.py (hunk 3 above)
Comment (olupton) [reply var/spack/repos/builtin/packages/perl-term-readkey/package.py]: Done, thanks for catching that!

**Review by alalazo — APPROVED** [review]

**Review by glennpj — APPROVED** [review]

**Pull request merged**